### PR TITLE
Ensure invoked Sphinx matches importable one

### PR DESCRIPTION
**Pull request by eric-wieser**
This should fix crashes when both conda and pip have Sphinx installed. It's likely also negligibly faster since it avoids the binary wrappers.

### CODE REVIEW

**Comment by eric-wieser** [comment]
Note that this approach is already what you use for mkdocs:

File: readthedocs/doc_builder/backends/sphinx.py [hunk 1]
```diff
@@ -232,9 +232,7 @@ def build(self):
         self.clean()
         project = self.project
         build_command = [
-            'python',
-            self.python_env.venv_bin(filename='sphinx-build'),
-            '-T',
+            'python', '-m', 'sphinx', '-T',
```
Comment (stsewd) [inline readthedocs/doc_builder/backends/sphinx.py:235]: This is missing around `python`

https://github.com/readthedocs/readthedocs.org/blob/fb01c6d9d78272e3f4fd146697e8760c04e4fbb6/readthedocs/doc_builder/backends/mkdocs.py#L271-L271

Comment (eric-wieser) [reply readthedocs/doc_builder/backends/sphinx.py:235]: Good catch, I wonder if there are other places with the same problem.

Is it the case that this file is always run with the same python as the venv one? If so, can we use `sys.executable` here?

Comment (stsewd) [reply readthedocs/doc_builder/backends/sphinx.py:235]: @eric-wieser we don't call to `activate`, that's why we need to full binary path here.

Comment (eric-wieser) [reply readthedocs/doc_builder/backends/sphinx.py]: Fair enough. Updated.

**Comment by eric-wieser** [comment]
> We need to test this across all sphinx versions (>=1.5).
> 
> But looks like it should work https://github.com/sphinx-doc/sphinx/blob/v1.5/sphinx/__main__.py

It was added in 1.3: https://github.com/sphinx-doc/sphinx/commit/74c7a52eca1fe69e67d2da7881e625cf7e8aee85

File: readthedocs/rtd_tests/tests/test_builds.py [hunk 2]
```diff
@@ -71,7 +71,8 @@ def test_build(self, load_config):
         self.assertEqual(self.mocks.popen.call_count, 3)
         cmd = self.mocks.popen.call_args_list[2][0]
         self.assertRegex(cmd[0][0], r'python')
-        self.assertRegex(cmd[0][1], r'sphinx-build')
+        self.assertRegex(cmd[0][1], r'-m')
+        self.assertRegex(cmd[0][1], r'sphinx')
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_builds.py:75]:
```suggestion
        self.assertRegex(cmd[0][2], r'sphinx')
```

**Comment by ericholscher** [comment]
Interesting. Anything that changes how we do builds scares me, but this seems like a reasonable change. I wonder if we should put it behind a feature flag to test it, and then roll it out via `historical_default=True`, so it's easy to rollback if needed?

**Comment by stsewd** [comment]
@eric-wieser this is how we add feature flags https://github.com/readthedocs/readthedocs.org/pull/6729/files let me know if you want to give it a try o I can take it from here.

The flag could be called `USE_SPHINX_FROM_VENV` or `FORCE_SPHINX_FROM_VENV` or similar

**Comment by eric-wieser** [comment]
I'd prefer if you could take it from here, thanks.

I think a more suitable name for the flag might be `DONT_MIX_PYTHONS_FOR_SPHINX` - currently the system python is being used to invoke a script from the venv, which seems like a poor idea. The `-m sphinx` change here is probably far less relevant than the switch to using the venv python. If you're concerned, you could roll out the change in two pieces.

Whatever you decide, could you enable this new behavior on the `cocotb.rtfd.io` PR builds  (and ping me when you've done so), so we can see if it fixes our issue? Thanks!

File: readthedocs/projects/models.py [hunk 3]
```diff
@@ -1665,6 +1666,8 @@ def add_features(sender, **kwargs):
             _('Limit the amount of concurrent builds'),
         ),
         (
+            FORCE_SPHINX_FROM_VENV,
+            _('Force to use Sphinx from the current virtual environment'),
             LIST_PACKAGES_INSTALLED_ENV,
```
Comment (eric-wieser) [inline readthedocs/projects/models.py:1671]:
```suggestion
        ),
        (
            LIST_PACKAGES_INSTALLED_ENV,
```

Comment (eric-wieser) [reply readthedocs/projects/models.py:1671]: Failure seems like a latent issue elsewhere:
```
Exception occurred:
  File "/home/travis/build/readthedocs/readthedocs.org/docs/doc_extensions.py", line 52, in feature_flags_role
    return [desc], []
UnboundLocalError: local variable 'desc' referenced before assignment
```

**Comment by eric-wieser** [comment]
Should there be a test for this? There was one present in 1fa3dda2dead4e610f14a9a1434f40baf179c4b1, but it seems it was removed again.

**Comment by stsewd** [comment]
> Should there be a test for this? There was one present in 1fa3dda, but it seems it was removed again.

I'll add it back when the feature flag is enabled by default.

**Comment by humitos** [comment]
@eric-wieser I enabled this feature flag in your project. Let me know if it does work as you expected. Thanks!

**Comment by eric-wieser** [comment]
Definitely doesn't break anything. Not sure if I can easily reproduce the original issue, since it relied on PyPI having a newer release of sphinx than conda, something that only happens immediately after a sphinx release.

**Pull request merged**